From 13ae5855f23623dd56b8c535c35277bd128b36d0 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 17 Apr 2017 18:35:55 +0800
Subject: [PATCH 1705/5242] MLK-14703-2 video: mipi_dsi_northwest: record
 traffic-mode.

commit  554f82726f9c2528d2bbffe614739290c2abdebd from
https://source.codeaurora.org/external/imx/linux-imx.git

Add a new field 'traffic_mode' to struct 'mipi_dsi_info" to
record the traffic mode read from dts or set to default for
mipi dsi panel display.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit e5ffea789dc1cb00f291383b9d5927ab1bba501f)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.h           |    1 +
 drivers/video/fbdev/mxc/mipi_dsi_northwest.c |   11 ++++++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.h b/drivers/video/fbdev/mxc/mipi_dsi.h
index c3514d9..04008b3 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.h
+++ b/drivers/video/fbdev/mxc/mipi_dsi.h
@@ -72,6 +72,7 @@ struct mipi_dsi_info {
 	int				dsi_power_on;
 	int				lcd_inited;
 	int				encoder;
+	int				traffic_mode;
 	u32				dphy_pll_config;
 	int				dev_id;
 	int				disp_id;
diff --git a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
index 9ff5d92..d710a97 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
@@ -918,6 +918,13 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 		memcpy(mipi_dsi->mode, &mxc_cea_mode[vmode_index],
 		       sizeof(struct fb_videomode));
 
+		ret = of_property_read_u32(remote, "dsi-traffic-mode",
+					   &mipi_dsi->traffic_mode);
+		if (ret < 0 || mipi_dsi->traffic_mode > 2) {
+			devm_kfree(&pdev->dev, mipi_dsi->mode);
+			return -EINVAL;
+		}
+
 		mipi_dsi->lcd_config = devm_kzalloc(&pdev->dev,
 						sizeof(struct mipi_lcd_config),
 						GFP_KERNEL);
@@ -927,7 +934,9 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 		}
 
 		mipi_dsi->encoder = 1;
-	}
+	} else
+		/* Default, using 'BURST-MODE' for mipi panel */
+		mipi_dsi->traffic_mode = 2;
 
 	ret = of_property_read_string(np, "lcd_panel", &lcd_panel);
 	if (ret) {
-- 
1.7.9.5

